File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
 const operations = {
        '+': (a, b) => a + b,
        '-': (a, b) => a - b,
        '*': (a, b) => a * b,
        '/': (a, b) => Math.trunc(a / b)
      };
    
      const stack = [];
    
      for (let i of tokens) {
        if (operations[i] != null) {
          const b = stack.pop();
          const a = stack.pop();
          stack.push(operations[i](a, b));
        } else {
          stack.push(Number(i));
        }
      }
    
      return stack[0];
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution(object):
    def evalRPN(self, tokens):
        operations = {
            '+': lambda a, b: a + b,
            '-': lambda a, b: a - b,
            '*': lambda a, b: a * b,
            '/': lambda a, b: int(float(a) / b)
        }

        stack = []

        for i in tokens:
            if i in operations:
                b = stack.pop()
                a = stack.pop()
                stack.append(operations[i](a, b))
            else:
                stack.append(int(i))

        return stack[0]

